### ISSUE TITLE
Bug: [unbound-method] Does not specify which destructured method is unbound

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#6276]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.9.3&sourceType=module&showAST=ts&code=MYewdgzgLgBAbgQwDYFcCmEYF4YG8BQMRMCAFAJR4wC+ANIcQEYBcMF2AfFXQ9fvqEixcJWjEY1s8ZOggBuIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6WJgIwHsOATWgFtE+ABbc+6KImjRu0SODABfEMqA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
const values = {
    a() { },
    b: () => { },
  }

const { a, b } = values;
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/unbound-method": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

Specifically `a` should be indicated as the offending unbound method.

### Actual Result

All of `{ a, b }` is squigglied.

### Additional Info

Found when looking at tRPC's source. Thanks for the poke @katt 😄 

https://github.com/trpc/trpc/blob/269c9a2eedeae075be7f9863f870d907cb36d01d/examples/standalone-server/src/server.ts#L75-L78
